src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -299,15 +299,25 @@ public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithDataExplorer(t
             throw new NotSupportedException($"The Data Explorer endpoint is only available when using the preview version of the Azure Cosmos DB emulator. Call '{nameof(RunAsPreviewEmulator)}' instead.");
         }
 
-        return builder.WithEndpoint(endpointName: "data-explorer", endpoint =>
-        {
-            endpoint.UriScheme = "http";
-            endpoint.TargetPort = 1234;
-            endpoint.Port = port;
-        });
+        var endpointName = "data-explorer";
+        return
+            builder.WithEndpoint(endpointName: endpointName, endpoint =>
+            {
+                endpoint.UriScheme = "http";
+                endpoint.TargetPort = 1234;
+                endpoint.Port = port;
+            })
+            .WithUrls(context =>
+            {
+                var url = context.Urls.FirstOrDefault(u => u.Endpoint?.EndpointName == endpointName);
+                if (url is not null)
+                {
+                    url.DisplayText = Resources.DataExplorerUrlText;
```

Comment:
I don't think this will work, because the requesting browser may be using a different locale than hosting. I think it should go in the place I linked before

Reply:
Ah right. We need this to be localized in the dashboard itself then, as a known string?

Reply:
Right. I think we'll have to use a combo of resource name + endpoint name?

Reply:
OK rigged it up.